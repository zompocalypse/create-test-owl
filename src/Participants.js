import React from 'react';
import './participants.css'

export default function Participants(props) {
  return (
    <div className="participants">
      <div className="session-participant">
        <img src={props.avatar} alt={props.name}/>
        <div className="participant-details">
          <h6>{props.name}</h6>
          <div className="status">
          {props.inSession}
          </div>  
        </div>
      </div>
    </div>
  )
}
