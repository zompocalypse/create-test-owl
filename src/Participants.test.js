import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participants from './Participants';
import store from './store';

describe( 'Participant component' , () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participants />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<Participants
        name={store.participants[0].name}
        avatar={store.participants[0].avatar}
        inSession={store.participants[0].inSession}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})