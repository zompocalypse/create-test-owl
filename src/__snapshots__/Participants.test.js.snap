// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component renders UI as expected 1`] = `
<div
  className="participants"
>
  <div
    className="session-participant"
  >
    <img
      alt="Koren Templeton"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-details"
    >
      <h6>
        Koren Templeton
      </h6>
      <div
        className="status"
      />
    </div>
  </div>
</div>
`;

exports[`renders UI as expected 1`] = `
<div
  className="participants"
>
  <div
    className="session-participant"
  >
    <img
      alt="Koren Templeton"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-details"
    >
      <h6>
        Koren Templeton
      </h6>
      <div
        className="status"
      />
    </div>
  </div>
</div>
`;
